diagram-viewer/src/main/java/com/powsybl/diagram/viewer/nad/NetworkAreaDiagramModel.java

```diff
@@ -97,18 +96,27 @@ public LayoutParameters getLayoutParameters() {
         return layoutParametersBean.getLayoutParameters();
     }
 
-    public LabelProvider getLabelProvider(Network network) {
-        return DEFAULT_LABEL_PROVIDER.equals(labelProvider.getValue()) ? new DefaultLabelProvider(network, getSvgParameters()) : null;
+    public LabelProviderFactory getLabelProviderFactory() {
+        return DEFAULT_LABEL_PROVIDER.equals(labelProvider.getValue())
+                ? DefaultLabelProvider::new
+                : null;
     }
 
-    public StyleProvider getStyleProvider(Network network) {
+    public StyleProviderFactory getStyleProviderFactory() {
         return TOPOLOGICAL_STYLE_PROVIDER.equals(styleProvider.getValue())
-                ? new TopologicalStyleProvider(network)
-                : new NominalVoltageStyleProvider(network);
+                ? TopologicalStyleProvider::new
+                : NominalVoltageStyleProvider::new;
     }
 
-    public LayoutFactory getLayoutFactory() {
-        return BASIC_LAYOUT.equals(layoutFactory.getValue()) ? new BasicForceLayoutFactory() : null;
+    public LayoutFactory getLayoutFactory(Network network) {
+        // FIXME : user must be able to customize positions
+        Map<String, Point> initialPositions = new HashMap<>();
+        return switch (layoutFactory.getValue()) {
+            case BASIC_LAYOUT -> new BasicForceLayoutFactory();
```

Comment:
"basic", for "BasicForceLayoutFactory" is not very self-explanatory. I would name it FORCE_LAYOUT for example. And I would change the displayed value to from "Basic" to "Force Layout" (see suggestion below).